burr/integrations/persisters/postgresql.py

```diff
@@ -0,0 +1,234 @@
+try:
+    import psycopg2
+except ImportError:
+    raise ImportError(
+        "Missing plugin PostgreSQL! To use the PostgreSQL plugin, you must install the following libraries: psycopg2. "
+        "You can install this with burr[postgresql] or pip install psycopg2 (replace with your package manager of choice)."
+    )
+
+import json
+import logging
+from typing import Literal, Optional
+
+from burr.core import persistence, state
+
+logger = logging.getLogger(__name__)
+
+
+class PostgreSQLPersister(persistence.BaseStatePersister):
+    """Class for PostgreSQL persistence of state. This is a simple implementation.
+
+    To try it out locally with docker -- here's a command -- change the values as appropriate.
+
+    .. code:: bash
+
+        docker run --name local-psql \  # container name
+                   -v local_psql_data:/SOME/FILE_PATH/ \  # mounting a volume for data persistence
+                   -p 54320:5432 \  # port mapping
+                   -e POSTGRES_PASSWORD=my_password \  # superuser password
+                   -d postgres  # database name
+
+    Then you should be able to create the class like this:
+
+    .. code:: python
+
+        p = PostgreSQLPersister.from_values("postgres", "postgres", "my_password",
+                                           "localhost", 54320, table_name="burr_state")
+
+
+    """
+
+    @classmethod
+    def from_config(cls, config: dict) -> "PostgreSQLPersister":
```

Comment:
Why do we have this? Just postgres convention?

Reply:
I'd want it.